tests/test_cli/test_new_mnemonic.py

```diff
@@ -257,6 +375,128 @@ def mock_get_mnemonic(language, words_path, entropy=None) -> str:
     clean_key_folder(my_folder_path)
 
 
+def test_gnosis_new_mnemonic_compounding_custom_amount(monkeypatch) -> None:
+    # monkeypatch get_mnemonic
+    def mock_get_mnemonic(language, words_path, entropy=None) -> str:
+        return "abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon about"
+
+    monkeypatch.setattr(new_mnemonic, "get_mnemonic", mock_get_mnemonic)
+
+    # Prepare folder
+    my_folder_path = os.path.join(os.getcwd(), 'TESTING_TEMP_FOLDER')
+    clean_key_folder(my_folder_path)
+    if not os.path.exists(my_folder_path):
+        os.mkdir(my_folder_path)
+
+    custom_amount = 1050
+
+    runner = CliRunner()
+    withdrawal_address = '0x00000000219ab540356cBB839Cbe05303d7705Fa'
+    inputs = ['english', '1', 'gnosis', 'MyPasswordIs', 'MyPasswordIs', withdrawal_address, withdrawal_address,
+              'yes', str(custom_amount),
+              'abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon about', '']
+    data = '\n'.join(inputs)
+    arguments = [
+        '--language', 'english',
+        '--ignore_connectivity',
+        'new-mnemonic',
+        '--folder', my_folder_path,
+    ]
+    result = runner.invoke(cli, arguments, input=data)
+    assert result.exit_code == 0
+
+    # Check files
+    validator_keys_folder_path = os.path.join(my_folder_path, DEFAULT_VALIDATOR_KEYS_FOLDER_NAME)
+    _, _, key_files = next(os.walk(validator_keys_folder_path))
+
+    deposit_file = [key_file for key_file in key_files if key_file.startswith('deposit_data')][0]
+    with open(validator_keys_folder_path + '/' + deposit_file, 'r', encoding='utf-8') as f:
+        deposits_dict = json.load(f)
+    for deposit in deposits_dict:
+        withdrawal_credentials = bytes.fromhex(deposit['withdrawal_credentials'])
+        assert withdrawal_credentials == (
+            COMPOUNDING_WITHDRAWAL_PREFIX + b'\x00' * 11 + decode_hex(withdrawal_address)
+        )
+        amount = deposit['amount']
+        assert amount == (custom_amount * 32 * ETH2GWEI)
+
+    all_uuid = [
+        get_uuid(validator_keys_folder_path + '/' + key_file)
+        for key_file in key_files
+        if key_file.startswith('keystore')
+    ]
+    assert len(set(all_uuid)) == 1
+
+    # Verify file permissions
+    if os.name == 'posix':
+        for file_name in key_files:
+            assert get_permissions(validator_keys_folder_path, file_name) == '0o400'
+
+    # Clean up
+    clean_key_folder(my_folder_path)
+
+
+def test_chiado_new_mnemonic_compounding_custom_amount(monkeypatch) -> None:
```

Comment:
Add another test here with a custom `--devnet_chain_setting` flag with custom and uncommon multiplier and min_deposit_amount for an non-existing network to test that this workflow is good.

Reply:
Multiple tests with a custom `--devnet_chain_setting` would be even better. One with the correct workflow where the deposit amount is valid keys generated. One with `amount_gwei < min_amount` and it fails because of that. One with `amount_gwei > MAX_DEPOSIT_AMOUNT * chain_setting.MULTIPLIER` and it fails because of that.